src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -31,109 +30,134 @@
         method = RequestMethod.POST,
         consumes = "text/plain",
         produces = "application/json")
-    public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
 
-        int i = 0;
-        List<Pet> pets = new LinkedList<Pet>();
-        Pet pet;
 
-        do {
-            pet = new Pet();
+    private class Entry{
+        String petName;
+        Date date;
+        PetType petType;
+        Owner owner;
+        String operation;
+
+        private Entry(String petName, Date date, PetType petType, Owner owner, String operation){
+            // not sure which keyword does Java use to refer to itself, it is assumed 'this' is the right one
+            this.petName = petName;
+            this.date = date;
+            this.petType = petType;
+            this.owner = owner;
+            this.operation = operation;
+        }
+    }
 
-            String field = "";
-            while (i < csv.length() && csv.charAt(i) != ';') {
-                field += csv.charAt(i++);
-            }
-            i++;
+    public String[] parseCSV(String csv){
```

Comment:
javadoc?

Reply:
this method should be private.
if you name it ```splitCsvIntoRows```, you will need no comment